### PR TITLE
Fix ui graph decimal axis

### PR DESCRIPTION
**Pull request by yamatoya**
#3084 

add format
* /1e-9 : n
* /1e-6 : µ
* /1e-3 : m

### CODE REVIEW

File: web/ui/static/js/graph.js [hunk 1]
```diff
@@ -733,6 +733,31 @@ Prometheus.Graph.prototype.remove = function() {
   self.handleChange();
 };
 
+Prometheus.Graph.prototype.formatKMBT = function(y) {
+  var abs_y = Math.abs(y);
+  if (abs_y >= 1e12) {
+    return y / 1e12 + "T"
```
Comment (stuartnelson3) [inline web/ui/static/js/graph.js]: Could you add `;` after each one of these returns?

File: web/ui/static/js/graph.js [hunk 2]
```diff
@@ -733,6 +733,31 @@ Prometheus.Graph.prototype.remove = function() {
...
+  } else if (abs_y >= 1e9) {
+    return y / 1e9 + "B"
+  } else if (abs_y >= 1e6) {
+    return y / 1e6 + "M"
+  } else if (abs_y >= 1e3) {
+    return y / 1e3 + "K"
```
Comment (stuartnelson3) [inline web/ui/static/js/graph.js]: the `kilo` prefix is represented by a small `k` (instead of `K`)

File: web/ui/static/js/graph.js [hunk 3]
```diff
@@ -733,6 +733,31 @@ Prometheus.Graph.prototype.remove = function() {
   self.handleChange();
 };
 
+Prometheus.Graph.prototype.formatKMBT = function(y) {
+  var abs_y = Math.abs(y);
+  if (abs_y >= 1e12) {
+    return y / 1e12 + "T"
+  } else if (abs_y >= 1e9) {
+    return y / 1e9 + "B"
```
Comment (stuartnelson3) [inline web/ui/static/js/graph.js]: What does `B` represent? Should it be `G`, for giga?

**Comment by beorn7** [comment]
Let's just be complete and consistent and take all the prefixes from https://en.wikipedia.org/wiki/Metric_prefix with the exception of those with exponents not dividable by 3 (i.e. d, c, da, h)

**Comment by yamatoya** [comment]
@stuartnelson3 
@beorn7 

thank you and done.

**Comment by beorn7** [comment]
:+1: But I'll wait for @stuartnelson3 's approval.

**Comment by stuartnelson3** [comment]
Code is good, but a small visual preference question:

The larger units currently don't have any cap to their length (unlike the smaller units with the negative exponents), so:

```javascript
y = 1234486
1234486
y / 1e6
1.234486
y / 1e6 + "M"
"1.234486M"
(y / 1e6).toFixed(3) + "M"
"1.234M"
```

And this gets increasingly silly:

```javascript
y = 1234567543478654
1234567543478654
y / 1e15
1.234567543478654
(y / 1e15).toFixed(3)
"1.235"
```

@beorn7 should we do an "across the board" limit of decimal accuracy to 3 places?

**Comment by grobie** [comment]
SGTM. The exact values will be displayed in the graph hover box anyway.

**Comment by beorn7** [comment]
We should limit the number of digits everywhere to the same value, but perhaps more than 3? How much does fit into the space in the graph? The fewer digits we allow, the more likely we run into a similar case as before, where the lower bar is 1.2340 and the upper bar is 1.2344, and both display as 1.234.

**Comment by stuartnelson3** [comment]
Your call, but I think 5 is probably the most we should do for this first attempt. I'm not sure how many numbers the container/graph shows well.

**Comment by beorn7** [comment]
I think we should have 5 consistently.

I can change that while fixing the merge conflict. Will merge into release-1.7, canary, and then draft a release.

**Comment by beorn7** [comment]
So this is a bit more complicated than anticipated. With the `toFixed`, it will display the given number of digits in any case, even if it is a perfectly round number. The plain old `toString` does a quite good job.
I'll add this to this PR while merging to save a roundtrip. It's also just being merge into a branch.

**Comment by beorn7** [comment]
Please check out what we have now in the release-1.7 branch. I think that's pretty much what we wanted.

**Pull request merged**